experiment_pages/experiment/map_rfid.py

```diff
@@ -11,142 +11,257 @@
+        )
+        simulate_rfid_button.place(relx=0.80, rely=0.17, anchor=CENTER)
+
+        # Simulate All RFID Button
+        simulate_all_rfid_button = CTkButton(
+            self, text="Simulate All RFID", compound=TOP, width=15, 
+            command=self.simulate_all_rfid
+        )
+        simulate_all_rfid_button.place(relx=0.80, rely=0.27, anchor=CENTER)
+
+    def add_value(self, rfid):
+        '''Adds an RFID entry to the table and updates animals list.'''
+        try:
+            item_id = len(self.animals) + 1  # Increment animal ID
+            self.animals.append({'id': item_id, 'rfid': rfid})  # Update animals list
+
+            print(f"Inserting into table: Animal {item_id}, RFID {rfid}")
+            self.table.insert('', 'end', values=(item_id, rfid))  # Insert into table
+
+            self.change_entry_text()  # Update the animal ID entry text
+            print("Table insertion and entry text update successful.")
+
+        except Exception as e:
+            print(f"Error in add_value: {e}")
 
-        self.menu_button.configure(command = self.press_back_to_menu_button)
+
+    def change_entry_text(self):
+        '''Updates the animal ID entry text to the next unmapped animal.'''
+        next_animal = len(self.animals) + 1  # Calculate the next animal ID
+        self.animal_id_entry_text.set(str(next_animal))  # Update the entry text
+
+
+    def add_random_rfid(self, play_sound=True):
+        '''Adds a random RFID value to the next animal.'''
+        try:
+            print(f"Attempting to add RFID. Current animal count: {len(self.animals)}")
+
+            if len(self.animals) >= self.db.get_number_animals():
+                self.raise_warning("All animals are already mapped.")
+                return
+
+            rfid = get_random_rfid()
+            print(f"Generated RFID: {rfid}")
+
+            # Schedule the addition using after() to avoid race conditions
+            self.master.after(0, lambda: self.add_value(rfid))
+            print(f"RFID {rfid} added successfully for animal {len(self.animals)}")
+
+            # Play audio if enabled, with a small delay to avoid threading issues
+            if play_sound:
+                self.master.after(100, lambda: AudioManager.play("shared/sounds/rfid_success.wav"))
+
+        except Exception as e:
+            print(f"Error in add_random_rfid: {e}")
+
+
+
+
+
+    def simulate_all_rfid(self):
+        '''Simulates RFID values for all unmapped animals with increased delay.'''
+        try:
+            unmapped_count = self.db.get_number_animals() - len(self.animals)
+
+            if unmapped_count <= 0:
+                self.raise_warning("All animals are already mapped.")
+                return
+
+            # Use a larger delay between each operation to prevent overload
+            for i in range(unmapped_count):
+                self.master.after(i * 100, self.add_random_rfid, False)  # 100ms delay per animal
```

Comment:
Why not call the individual simulate rfid method rather than trying to reinvent the wheel?

Reply:
Has this been tried yet?